src/items.rs

```diff
@@ -1357,10 +1357,21 @@ fn rewrite_fn_base(context: &RewriteContext,
             FnArgLayoutStyle::Block if put_args_in_block => false,
             FnArgLayoutStyle::BlockAlways => false,
             _ => {
-                // If we've already gone multi-line, or the return type would push
-                // over the max width, then put the return type on a new line.
-                result.contains("\n") || multi_line_ret_str ||
-                result.len() + indent.width() + ret_str_len > context.config.max_width
+                // If we've already gone multi-line, or the return type would push over the max
+                // width, then put the return type on a new line. With the +1 for the signature
+                // length an additional space between the closing brace of the argument and the
+                // arrow '->' is considered.
+                let mut sig_length = result.len() + indent.width() + ret_str_len + 1;
+
+                // If there is no where clause, take into account the space after the return type
+                // and the brace.
+                if where_clause.predicates.is_empty() {
```

Comment:
Oh, I overlooked that... :sweat_smile: but to be sure I will add a test for that case later.